### PR TITLE
[qa] Add optional unique coinbase generation

### PR DESCRIPTION
**Pull request by venatici**
Optionally generate unique coinbase with the same height and pubkey.

Alternative to #7886 maintaining backwards compat.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Can you explain why it is a good thing to keep backward compat in this very case?

File: qa/rpc-tests/test_framework/blocktools.py [hunk 1]
```diff
@@ -40,10 +40,18 @@ def serialize_script_num(value):
...
 # If pubkey is passed in, the coinbase output will be a P2PK output;
 # otherwise an anyone-can-spend output.
-def create_coinbase(height, pubkey = None):
+def create_coinbase(height, pubkey = None, unique = False):
     coinbase = CTransaction()
-    coinbase.vin.append(CTxIn(COutPoint(0, 0xffffffff), 
-                ser_string(serialize_script_num(height)), 0xffffffff))
+    scriptSig = ser_string(serialize_script_num(height))
+
+    if (unique == True):
```
Comment (dhimmel) [inline qa/rpc-tests/test_framework/blocktools.py]: Would this line be better as either:

```python
if unique:
```

or

```python
if unique is None:
```

Comment (jtimon) [inline qa/rpc-tests/test_framework/blocktools.py]:
```
if unique:
```
Is the simplest form.

**Comment by jtimon** [comment]
Concept ACK

**Comment by laanwj** [comment]
Nothing uses the new functionality of `create_coinbase`?

**Comment by mrbandrews** [comment]
I don't feel strongly about it, but I probably prefer the simplicity of 7886.  I think the caller's usual expectation would be to get a unique coinbase, even for the same height and same pubkey (which would be reused for convenience, not because the test code is thinking of a single miner producing all the blocks).  I verified the assertion in 7886 that it doesn't break any of the existing tests.

**Comment by laanwj** [comment]
Closing this - There is no point in merging this as-is. if anyone needs this functionality for a test they are writing, they can cherry-pick this as well.

**Pull request closed without merging**